woocommerce/woocommerce-ios
# Add banner to purchase a paid plan for WooExpress sites with expired plan

Closes: #10298


## Description

As discussed in pe5sF9-1JH-p2, when a store plan expires, after 8 days the site gets reverted to a simple site and the app is no longer usable. To mitigate this issue, we should detect these cases and let users upgrade their plans in the app. 

Previously, we added an alert with a CTA to upgrade plans if this happens in #10059 - but this alert can be dismissed. For a more persistent solution, we should make use of the existing free trial banner to display something similar on the app.

This PR makes use of the existing `FreeTrialBannerPresenter` by renaming it to `StorePlanBannerPresenter` to be more generic and catching the case where a store plan is reverted to simple while it is recognized as having run a free trial WooExpress plan in the past. We only support upgrading for WooExpress stores for now since IAP hasn't been implemented to support regular WPCom plans.

## Testing instructions

Prerequisite: Make sure that you have access to an expired WooExpress site.
- Expired sites are not accessible on the store picker as we are filtering them out. Build and run the code from [this diff](https://github.com/woocommerce/woocommerce-ios/compare/try/allow-ineligible-sites-in-picker?expand=1) to get access to those sites.
- In the store picker, select the expired WooExpress site and tap Continue. 
- You should be navigate to the dashboard screen with the banner saying the past plan has ended.
- The banner CTA should display "Learn More" if your account is ineligible for IAP, or "Upgrade" otherwise.

Please feel free to test with other sites to make sure that the new banner is only available for sites with expired WooExpress plans, and the free trial banner is still displayed for free trial stores.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/b3334d0a-e2c3-4e5a-8680-c1bf9c619c99" width=320 />



---
